File: tasks/implementations/java/tasks/ai_service_integration.java
import java.nio.file.*;
import java.util.regex.*;

class AiServiceIntegration {
    public static void run(String size, String fixturesRoot) throws Exception {
        String text = Files.readString(Path.of(fixturesRoot, "mocks", "ai_service_" + size + ".json"));
        Matcher modelMatcher = Pattern.compile("\\\"model\\\":\\s*\\\"([^\\\"]+)\\\"").matcher(text);
        modelMatcher.find();
        long total = modelMatcher.group(1).length();
        Matcher matcher = Pattern.compile("\\\"prompt\\\":\\s*\\\"([^\\\"]+)\\\",\\s*\\\"output\\\":\\s*\\\"([^\\\"]+)\\\",\\s*\\\"tokens\\\":\\s*(\\d+)").matcher(text);
        while (matcher.find()) total += matcher.group(1).length() + matcher.group(2).length() + Integer.parseInt(matcher.group(3));
        System.out.println(total);
    }
}

File: tasks/implementations/python2/tasks/ai_service_integration.py
import re

from tasks.compat import emit, join_path, read_text


def run(size, fixtures_root):
    text = read_text(join_path(fixtures_root, 'mocks', 'ai_service_%s.json' % size))
    model = re.search(r'"model":\s*"([^"]+)"', text).group(1)
    total = len(model)
    for prompt, output, tokens in re.findall(r'"prompt":\s*"([^"]+)",\s*"output":\s*"([^"]+)",\s*"tokens":\s*(\d+)', text):
        total += len(prompt) + len(output) + int(tokens)
    emit(total)
